internal/extender/overhead.go

```diff
@@ -39,151 +37,165 @@ var (
-		for _, podName := range rr.Status.Pods {
-			podsWithRRs[podName] = true
-		}
+// computeNodeOverhead adds the requests of pods that don't have reservations and are counted as overhead.
+// This returns (overhead, nonSchedulableOverhead).
+func (o *OverheadComputer) computeNodeOverhead(ctx context.Context, nodeName string) (*resources.Resources, *resources.Resources) {
+	o.overheadLock.RLock()
+	defer o.overheadLock.RUnlock()
+	nodeRequests, ok := o.resourceRequests[nodeName]
+	if !ok {
+		return resources.Zero(), resources.Zero()
 	}
-	rawOverhead := map[string]resources.NodeGroupResources{}
-	rawNonSchedulableOverhead := map[string]resources.NodeGroupResources{}
-	for _, p := range pods {
-		if podsWithRRs[p.Name] {
-			continue
-		}
-		if role, ok := p.Labels[common.SparkRoleLabel]; ok {
-			if role == common.Executor && o.softReservationStore.ExecutorHasSoftReservation(ctx, p) {
-				continue
-			}
-		}
-		if p.Spec.NodeName == "" || p.Status.Phase == v1.PodSucceeded || p.Status.Phase == v1.PodFailed {
-			// pending pod or pod succeeded or failed
-			continue
-		}
-
-		instanceGroup, err := o.getPodNodeInstanceGroup(p)
+	overhead := resources.Zero()
+	nonSchedulableOverhead := resources.Zero()
+	for _, podRequestInfo := range nodeRequests {
+		pod, err := o.podInformer.Lister().Pods(podRequestInfo.podNamespace).Get(podRequestInfo.podName)
 		if err != nil {
-			svc1log.FromContext(ctx).Warn("could not get instance group for node where pod is running, skipping", svc1log.SafeParam("failedPod", p.Name), svc1log.Stacktrace(err))
+			svc1log.FromContext(ctx).Warn("error when checking if pod has a reservation, node overhead calculation might be inaccurate",
+				svc1log.SafeParam("nodeName", nodeName),
+				svc1log.SafeParam("podName", podRequestInfo.podName),
+				svc1log.SafeParam("podNamespace", podRequestInfo.podNamespace))
 			continue
 		}
+		if !o.resourceReservationManager.PodHasReservation(ctx, pod) {
+			overhead.Add(podRequestInfo.requests)
+			if pod.Spec.SchedulerName != common.SparkSchedulerName {
+				nonSchedulableOverhead.Add(podRequestInfo.requests)
+			}
 
-		// found pod with no associated resource reservation, add to overhead
-		o.addPodResourcesToGroupResources(ctx, rawOverhead, p, instanceGroup)
-
-		if p.Spec.SchedulerName != common.SparkSchedulerName {
-			// add all pods that this scheduler does not deal with to the non-schedulable overhead
-			o.addPodResourcesToGroupResources(ctx, rawNonSchedulableOverhead, p, instanceGroup)
+			if podRequestInfo.requests.CPU.AsDec().Cmp(oneCPU.AsDec()) > 0 || podRequestInfo.requests.Memory.AsDec().Cmp(oneGiB.AsDec()) > 0 {
+				svc1log.FromContext(ctx).Debug("Container with no resource reservation has high resource requests",
+					svc1log.SafeParam("podName", pod.Name),
+					svc1log.SafeParam("nodeName", pod.Spec.NodeName),
+					svc1log.SafeParam("CPU", podRequestInfo.requests.CPU),
+					svc1log.SafeParam("Memory", podRequestInfo.requests.Memory))
+			}
```

Comment:
I believe this will become too spammy if we start logging it here on every check to get overhead, so we should either remove it if it doesn't provide a lot of value, or consider logging it differently.

Reply:
yea I think we can remove this log line